cmake-modules/DefineTransportAdapter.cmake

```diff
@@ -0,0 +1,29 @@
+## Copyright (c) Microsoft Corporation. All rights reserved.
+## SPDX-License-Identifier: MIT
+
+#############                      TRANSPORT ADAPTER BUILD                     #####################
+#  Default: If no option is explicitly added, curl will be used for POSIX and WIN HTTP for WIndows #
+#  Windows: Both CURL and WIN_HTTP can be build to be used.                                        #
+#  POSIX: Only CURL is acceptable. If WIN_HTTP is set, generate step will fail for user            #
+
+#  Defines `BUILD_TRANSPORT_WINHTTP_ADAPTER` and `BUILD_CURL_HTTP_TRANSPORT_ADAPTER` for source code
+
+if (WIN32 OR MINGW OR MSYS OR CYGWIN)
+  if(BUILD_TRANSPORT_CURL)
+    add_compile_definitions(BUILD_CURL_HTTP_TRANSPORT_ADAPTER)
+  endif()
+  # If explicitly adding WIN_HTTP or did not add CURL
+  if(BUILD_TRANSPORT_WINHTTP OR NOT BUILD_TRANSPORT_CURL)
```

Comment:
Please update the comment to explain "why are we doing this" or remove it.
 Re-stating what is obvious from the self-documenting if-check is unnecessary.